services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -267,58 +307,76 @@ async def lock_with_notification(
             user_id, project_uuid, app
         )
         log.error(
-            "Project [%s] already locked in state '%s'. Please check with support.",
+            "Project [%s] for %s already locked in state '%s'. Please check with support.",
             f"{project_uuid=}",
+            f"{user_id=}",
             f"{prj_states.locked.status=}",
         )
         raise
+
     finally:
         if notify_users:
+            # notifies as lock is released
             await retrieve_and_notify_project_locked_state(user_id, project_uuid, app)
 
 
 async def remove_project_interactive_services(
-    user_id: int,
+    user_id: Optional[int],  # Not guaranteed to exist
     project_uuid: str,
     app: web.Application,
     notify_users: bool = True,
-    user_name: Optional[UserNameDict] = None,
 ) -> None:
-    # NOTE: during the closing process, which might take awhile,
-    # the project is locked so no one opens it at the same time
+    """
+    raises ProjectNotFoundError
+    raises UserNotFoundError
+    """
+
     log.debug(
-        "removing project interactive services for project [%s] and user [%s]",
-        project_uuid,
-        user_id,
+        "Removing project interactive services for %s and %s and %s",
+        f"{project_uuid=}",
+        f"{user_id=}",
+        f"{notify_users=}",
     )
-    try:
-        async with lock_with_notification(
+
+    # can raise User
+    user_name_data = await get_user_name(app, user_id)
+    user_role: UserRole = await get_user_role(app, user_id)
+
+    with suppress(ProjectLockError):
```

Comment:
I think you are not using this correctly. If an error occurs you will exit the context manager. [See docs](https://docs.python.org/3/library/contextlib.html#contextlib.suppress)

Please also run the below example. Only `first` is being printed.

```python
import os
from contextlib import suppress

with suppress(FileNotFoundError):
    print("first")
    os.remove("hso.asdasda.ass")
    print("second")
```

If the user cannot be locked, the second suppress block where `director_v2_api.stop_dynamic_services_in_project` is called still has to run.

Reply:
It is used as intended. This execption is raised by the context manager ``lock_project_and_notify_state_update``. The logic is something like
```python
with suppress(ProjectLockError):
    with lock_project_and_notify_state_update():
          print("If the project is not lock, run me, otherwise skip")
```